### ISSUE TITLE
BraceWrapping is uneditable

### ISSUE THREAD

**Issue by tzahig** [zed0/clang-format-configurator#13]
`BraceWrapping` is set as "Default" and is uneditable. The frontend should be able to handle the nested configuration options.

**Comment by DamianSuess** [comment]
Same here. I tried a workaround by uploading, however, it throws the error "_The file you uploaded does not appear to be a valid YAML file_"

It appears to throw the exception when the following lines below are present. It's a shame too because these lines are critical to our formatting rules.

A good example for a .clang-format file that throws these false-positive errors can be found at https://github.com/intel/parameter-framework/blob/master/.clang-format

```
# Configure each individual brace in BraceWrapping
BreakBeforeBraces: 'Custom'

# Control of individual brace wrapping cases
BraceWrapping:
    AfterClass: true
    AfterControlStatement: false
    AfterEnum: true
    AfterFunction: true
    AfterNamespace: true
    AfterStruct: true
    AfterUnion: true
    BeforeCatch: false
    BeforeElse: false
    IndentBraces: false
```

***Updated with @Pharap's syntax suggestions***

**Comment by Pharap** [comment]
@DamianSuess 
In your case the reason for the error is because you're using curly braces.
From what I gather, YAML only allows curly braces for inline lists.
Also you've quoted what are supposed to be boolean values.

Perhaps whatever system you're working with is using a non-standard variant of YAML?

---

Either way, this bug is still a valid issue.

**Comment by DamianSuess** [comment]
@Pharap, thanks for the heads up! I updated the previous post to reflect your suggestions & it no longer throws an error. And yes, as you stated as well, it's still a bug.

Btw, we're using it in Visual Studio 2017 (v15.7.4) which worked. Surprised the included link w/ the wrong format was being used by Intel.

**Comment by Pharap** [comment]
Without meaning to derail the issue too much...

@DamianSuess I just double checked the spec and it seems that YAML does permit braces to be used in such a way and has done for some time.
(However the lack of commas in intel's example appears to be non-standard, so it's possible that their parser either has a bug or a non-standard feature.)

The spec for YAML versions 1.0, 1.1 and 1.2 all give the following as example 2.6:
```yaml
Mark McGwire: {hr: 65, avg: 0.278}
Sammy Sosa: {
    hr: 63,
    avg: 0.288
  }
```

In light of this I decided to test:
```YAML
# Configure each individual brace in BraceWrapping
BreakBeforeBraces: 'Custom'

# Control of individual brace wrapping cases
BraceWrapping: {
    AfterClass: true,
    AfterControlStatement: false,
    AfterEnum: true,
    AfterFunction: true,
    AfterNamespace: true,
    AfterStruct: true,
    AfterUnion: true,
    BeforeCatch: false,
    BeforeElse: false,
    IndentBraces: false
  }
```

With clang-format configurator it didn't work, so I believe you have indeed found a bug related to YAML parsing.

However, I dug into it a bit more and it seems the issue is actually with Jérémy Faivre's (@jeremyfa) [yaml.js](https://github.com/jeremyfa/yaml.js).
It has been raised as [issue #28](https://github.com/jeremyfa/yaml.js/issues/28).

So rather than raising an issue here, it would be better to bring attention to the issue on yaml.js, get it fixed there, and then clang-format configurator can just update its library and the problem should be fixed.

**Comment by zed0** [comment]
I think this is closed by @Skim-x-CAT 's changes in #16 which should be live now.  Let me know if you have any more issues.
